pkg/network/protocols/events/configuration.go

```diff
@@ -154,24 +145,14 @@ var noopInstruction = asm.Instruction{
 // essentially replace `bpf_ringbuf_output` helper calls by a noop operation so
 // they don't result in verifier errors even when deadcode elimination fails.
 func removeRingBufferHelperCalls(m *manager.Manager) {
-	m.InstructionPatchers = append(m.InstructionPatchers, func(m *manager.Manager) error {
-		progs, err := m.GetProgramSpecs()
-		if err != nil {
-			return err
-		}
-
-		for _, p := range progs {
-			iter := p.Instructions.Iterate()
-			for iter.Next() {
-				ins := iter.Ins
-				if ins.IsBuiltinCall() && ins.Constant == int64(asm.FnRingbufOutput) {
-					*ins = noopInstruction
-				}
-			}
-		}
-
-		return nil
-	})
+	// TODO: this is not the intended API usage of a `ebpf.Modifier`.
+	// Once we have access to the `ddebpf.Manager`, add this modifier to its list of
+	// `EnabledModifiers` and let it control the execution of the callbacks
+	patcher := ddebpf.NewHelperCallRemover(asm.FnRingbufOutput)
+	err := patcher.BeforeInit(m, nil)
+	if err != nil {
+		log.Errorf("error patching eBPF bytecode: %s", err)
```

Comment:
Do we not want to fail here?
If the patching failed, the verifier would likely complain.

Reply:
I actually think that not failing may be the right thing to do here.
The USM "production" code was actually loading on all Kernels we test on CI (including those that don't support Ring Buffers), presumably due to pruning/dead code elimination.
The only thing failing to load was actually a small eBPF *test* program. So I added the patching almost as an extra safety layer.

All that to say that even if the patching fails, I think there's still a good chance that the program will succeed to load. If it doesn't, I don't think there is much we can do, and the loading error will bubble up and be appropriate handled by the upstream code.

Reply:
Sounds good. Maybe adding this as a comment would be useful for down the road.

Reply:
Done!